### ISSUE TITLE
Completion incorrectly infers lambda parameter types when the lambda is passed as a named argument in a different position than the corresponding parameter

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#40216]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/825326/vs2019-intellisense-wrong.html)._

---
I have a post on stackoverflow a bout this problem. Please review it
https://stackoverflow.com/questions/58926131/graphql-netcore-visualstudio-2019-wrong-suggestions

---
#### Visual Studio Feedback System on 11/18/2019, 10:25 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Visual Studio Feedback System on 12/3/2019, 05:11 PM: 

<p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>

---
### Original Solutions
(no solutions)

**Comment by dpoeschl** [comment]
Found a minimal standalone repro:

```C#
using System;

class C
{
    void Test()
    {
        X(1, bar: t => Console.WriteLine(t.Length)); // Good: Length is in the list
        X(bar: t => Console.WriteLine(t.Length)); // Bad: Length is not in the list
    }

    void X(int foo = 7, Action<string> bar = null) { }
}
```

The code that figures out the possible types of lambda parameters assumes the index of the argument is the same as the index of the parameter, which is the case in `// Good` above, but not the case in `// Bad`. Here's where things go wrong in the recommender:

https://github.com/dotnet/roslyn/blob/c2cc40e3a86a9b9acac532cf497c5119a7afb18d/src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationServiceRunner.cs#L118

**Comment by giangcoi48k** [comment]
The issue has been closed, but the problem has not been resolved on the latest version 16.4.2 release today

**Comment by jmarolf** [comment]
@giangcoi48k this will be fixed in 16.5
